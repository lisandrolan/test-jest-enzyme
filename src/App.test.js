import React from 'react';

import {mount, shallow} from 'enzyme';
import App from './App';

// it('renders without crashing', () => {   const div =
// document.createElement('div');   ReactDOM.render(<App />, div);
// ReactDOM.unmountComponentAtNode(div); });

describe('App', () => {
  let app;
  it('should render correctly in "debug" mode', () => {
    app = shallow(<App debug/>);

    expect(app).toMatchSnapshot();
  });
  it('should render correctly with no props', () => {
     app = shallow(<App/>);

    expect(app).toMatchSnapshot();
  });
  it('App renders nested components', () => {
    expect(app.find('Add').length).toEqual(1);
    expect(app.find('List').length).toEqual(1);
  });
  it('onAdd updates List', () => {
    const add = app.find('Add').first();
    add.props().onAdd('Name 1');
    app.update();     
    const list = app.find('List').first();
    const listData = list.props().data;
    expect(listData.length).toEqual(1);
    expect(listData[0]).toEqual('Name 1');
  });
  // it('should render banner text correctly with given strings', () => {
  //   const strings = ['one', 'two'];
  //   const component = shallow(<App list={strings}/>);
  //   expect(component).toMatchSnapshot();
  // });
  // it('should be possible to open menu with Spacebar', done => {
  //   const component = mount(<App/>);
  //   component
  //     .find('#link-id')
  //     .simulate('keydown', {keyCode: 32});
  //   expect(component).toMatchSnapshot();
  //   component.unmount();
  // });
  // it('sets loading state to true on save press', () => {
  //   const component = mount(<App />);
  //   component.find('[className="save-button"]').simulate('click');
  //   expect(component.state('isLoading')).toEqual(true);
  //   component.unmount();
  // });
});