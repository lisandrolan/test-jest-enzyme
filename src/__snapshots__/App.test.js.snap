// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `
<div>
  <Add
    onAdd={[Function]}
  />
  <List
    data={Array []}
  />
</div>
`;

exports[`App should render correctly with no props 1`] = `
<div>
  <Add
    onAdd={[Function]}
  />
  <List
    data={Array []}
  />
</div>
`;
